src/indexer.ts

```diff
@@ -0,0 +1,76 @@
+export const indexer = {
+  buildIndexFromObject: function (
+    object: { readonly [x: string]: any } | undefined,
```

Comment:
are all the type annotations here really necessary? it seems excessive here, personally it reduces readability of the code